### PR TITLE
docs: info about cli log, run, status and update

### LINKED ISSUE #179: document dos/dont's for renku run
Better document what is allowed vs not allowed in the run command, pipes and process redirection with `<(...)` are not. Show an example of a pipe and explain how to potentially split it into two commands.

### PR DESCRIPTION
**Pull request by jirikuncar**
(closes #179)

### CODE REVIEW

File: renku/cli/log.py [hunk 1]
```diff
@@ -15,24 +15,60 @@
...
+
+Unlike the traditional file history format, which shows previous revisions
+of the file, this format presents tool inputs together with their revision
+identifiers.
+
+A ``*`` character shows to which lineage the specific file belongs to.
+A ``@`` character in the graph lineage means that the corresponding file does
+not have any inputs and the history starts there.
+
+When called without file names, it shows the history of latest created files.
```
Comment (rokroskar) [inline renku/cli/log.py]: When called without file names, `renku log` shows...

File: renku/cli/log.py [hunk 2]
```diff
@@ -15,24 +15,60 @@
...
+With the ``--revision <refname>`` option the output is show as it was in the
```
Comment (rokroskar) [inline renku/cli/log.py]: is shown

File: renku/cli/log.py [hunk 3]
```diff
@@ -15,24 +15,60 @@
...
+specified revision.
+
+Provenance examples
+```````````````````
+
+``renku log --revision HEAD~5``
+   Show the history of files that have been created or modified 5 commits ago.
+
+``renku log B``
```
Comment (rokroskar) [inline renku/cli/log.py]: put this (simplest) example at the top

File: renku/cli/run.py [hunk 4]
```diff
@@ -15,7 +15,63 @@
...
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Track provenance of data created by executing programs."""
+"""Track provenance of data created by executing programs.
+
+Capture command line execution
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tracking exection of your command line script is done by simply adding
+``renku run`` command before the previous arguments. The command will detect
```
Comment (rokroskar) [inline renku/cli/run.py]: the ``renku run`` command

File: renku/cli/run.py [hunk 5]
```diff
@@ -15,7 +15,63 @@
...
+
+* arguments (flags),
+* string and integer options,
+* input files if linked to existing files in the repository,
```
Comment (rokroskar) [inline renku/cli/run.py]: should we mention that inputs need to be specified on the command line? Should be obvious but maybe it isn't?

File: renku/cli/run.py [hunk 6]
```diff
@@ -15,7 +15,63 @@
...
+* output files if modified or created while running the command.
+
+.. note:: If there were uncommitted changes then the command fails.
+   Check :program:`git status` to see details.
```
Comment (rokroskar) [inline renku/cli/run.py]: See ... for details.

File: renku/cli/run.py [hunk 7]
```diff
@@ -15,7 +15,63 @@
...
+
+Detecting input files
+~~~~~~~~~~~~~~~~~~~~~
+
+The arguments are identified as input file only if they path matches an
```
Comment (rokroskar) [inline renku/cli/run.py]: An argument is identified as an input file only if its path matches an existing file in the repository.

File: renku/cli/run.py [hunk 8]
```diff
@@ -15,7 +15,63 @@
...
+
+* If the file is modified during the execution, then it is stored as an output;
+
+* If the path points to a directory, then it is stored as a string option.
+
+Detecting output files
+~~~~~~~~~~~~~~~~~~~~~~
+
+Any file which is modified or created after the execution will be added as an
+output. If the program does not have any output file, you can specify the
```
Comment (rokroskar) [inline renku/cli/run.py]: If the program does not produce any outputs, you can...

File: renku/cli/run.py [hunk 9]
```diff
@@ -15,7 +15,63 @@
...
+and recorded in the tool specification when invoked by ``renku run cat < A``.
+
+Similarly, both redirects to standard output and standard error output can be
+done when invoking a command:
+
+.. code-block:: console
+
+   $ renku run grep "test" B > C 2> D
+
+.. note:: Detecting inputs and outputs from pipes ``|`` is not supported.
```
Comment (rokroskar) [inline renku/cli/run.py]: note --> warning?

File: renku/cli/update.py [hunk 10]
```diff
@@ -15,7 +15,71 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Update an existing file."""
+r"""Update outdated files created by "run" command.
```
Comment (rokroskar) [inline renku/cli/update.py]: by the "run"...

File: renku/cli/update.py [hunk 11]
```diff
@@ -15,7 +15,71 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The information about dependencies for each file in the repository is generated
+from information stored in the underlying Git repository.
+
+A minimal dependency graph is generated for each outdated file stored in the
+repository. It means that only the necessary steps will be executed and the
+workflow used to orchestrate these steps is stored in the repository.
+
+Assume the following history for the file ``H`` exists.
```
Comment (rokroskar) [inline renku/cli/update.py]: Assume that the...

File: renku/cli/run.py (hunk 5 above)
Comment (jirikuncar) [reply renku/cli/run.py]: I have tried to put more details in `Detecting input files`

**Pull request merged**